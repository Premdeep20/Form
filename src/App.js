import React, { Component } from 'react';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import TextField from '@material-ui/core/TextField';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import Button from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Container from '@material-ui/core/Container';
import FormHelperText from '@material-ui/core/FormHelperText';

import { withStyles } from '@material-ui/core/styles';

export class App extends Component {

  state = {
    data: {
      firstName: '',
      lastName: '',
      email: '',
      phoneNo: '',
      department: '',
    },
    error: {
      firstName: false,
      lastName: false,
      email: false,
      phoneNo: false,
      department: false,
    }
  }

  change = (event) => {
    const { id, value } = event.target;
    const { data, error } = this.state;
    if (id === 'firstName') {
      let err = !this.validateName(value);
      this.setState((prevState) => ({
        ...prevState,
        data: {
          ...data,
          firstName: value,
        },
        error: {
          ...error,
          firstName: err,
        }
      }))
    } else if (id === 'lastName') {
      let err = !this.validateName(value);
      this.setState((prevState) => ({
        ...prevState,
        data: {
          ...data,
          lastName: value,
        },
        error: {
          ...error,
          lastName: err,
        }
      }))
    } else if (id === 'email') {
      let err = !this.validateEmail(value);
      this.setState((prevState) => ({
        ...prevState,
        data: {
          ...data,
          email: value,
        },
        error: {
          ...error,
          email: err,
        }
      }))
    } else if (id === 'phoneNo') {
      let err = !this.validatePhone(value);
      this.setState((prevState) => ({
        ...prevState,
        data: {
          ...data,
          phoneNo: value,
        },
        error: {
          ...error,
          phoneNo: err,
        }
      }))
    } else {
      this.setState((prevState) => ({
        ...prevState,
        data: {
          ...data,
          department: value,
        },
        error: {
          ...error,
          department: false,
        }
      }))
    }
  }

  validateName = (value) => {
    const reg = /^[a-zA-Z]+$/;
    return reg.test(value);
  }

  validateEmail = (value) => {
    const reg = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
    return reg.test(value);
  }

  validatePhone = (value) => {
    const reg = /^[789]\d{9}$/;
    return reg.test(value);
  }

  handleSubmit = () => {
    const { firstName, lastName, email, phoneNo, department } = this.state.data;
    const { error } = this.state;
    if (firstName && lastName && email && phoneNo && department) {
      if (!error.firstName && !error.lastName && !error.email && !error.phoneNo && !error.department) {
        let datas = JSON.parse(localStorage.getItem('DATAS'));
        if (!datas) {
          datas = [];
        }
        datas.push(this.state.data);
        localStorage.setItem('DATAS', JSON.stringify(datas));
        this.setState({
          data: {
            firstName: '',
            lastName: '',
            email: '',
            phoneNo: '',
            department: '',
          }
        });
      }
    } else {
      this.setState({
        error: {
          ...error,
          firstName: !firstName,
          lastName: !lastName,
          email: !email,
          phoneNo: !phoneNo,
          department: !department,
        }
      })
    }
  }

  getData = (type) => {
    const { classes } = this.props;
    let datas = JSON.parse(localStorage.getItem('DATAS'));
    datas = datas.filter(data => data.department === type);
    return <>
      {
        (datas.length > 0) && <Typography variant="h5" gutterBottom align="center" className={classes.typography}>
          {type}
        </Typography>
      }
      {datas.map((data) => (
        <Grid item xs={12} sm={6} md={4}>
          <Card className={classes.card}>
            <CardContent className={classes.cardContent}>
              <Typography gutterBottom variant="h6">
                {data.firstName} {data.lastName}
              </Typography>
              <Typography color="textSecondary">
                {data.email}
              </Typography>
              <Typography color="textSecondary">
                {data.phoneNo}
              </Typography>
              <Typography color="textSecondary">
                {data.department}
              </Typography>
            </CardContent>
          </Card>
        </Grid>
      ))}
    </>
  }

  render() {
    const { classes } = this.props;
    const { firstName, lastName, email, phoneNo, department } = this.state.data;
    const datas = JSON.parse(localStorage.getItem('DATAS'));
    const { error } = this.state;
    return (
      <>
        <main className={classes.layout}>
          <Paper className={classes.paper}>
            <React.Fragment>
              <Typography variant="h6" gutterBottom>
                Employee details
            </Typography>
              <Grid container spacing={3}>
                <Grid item xs={12} sm={6}>
                  <TextField
                    id="firstName"
                    name="firstName"
                    label="First name"
                    value={firstName}
                    fullWidth
                    onChange={this.change}
                    error={error.firstName}
                    helperText={error.firstName && 'Please fill valid first name!'}
                  />
                </Grid>
                <Grid item xs={12} sm={6}>
                  <TextField
                    id="lastName"
                    name="lastName"
                    label="Last name"
                    value={lastName}
                    required
                    fullWidth
                    onChange={this.change}
                    error={error.lastName}
                    helperText={error.lastName && 'Please fill valid last name!'}
                  />
                </Grid>
                <Grid item xs={12}>
                  <TextField
                    id="email"
                    name="email"
                    label="Email"
                    value={email}
                    required
                    fullWidth
                    onChange={this.change}
                    error={error.email}
                    helperText={error.email && 'Please fill valid email!'}
                  />
                </Grid>
                <Grid item xs={12} sm={6}>
                  <TextField
                    id="phoneNo"
                    name="phoneNo"
                    label="Phone no"
                    value={phoneNo}
                    required
                    fullWidth
                    onChange={this.change}
                    error={error.phoneNo}
                    helperText={error.phoneNo && 'Please fill valid phone no!'}
                  />
                </Grid>
                <Grid item xs={12} sm={6}>
                  <FormControl
                    variant="outlined"
                    required
                    fullWidth
                    className={classes.formControl}
                    error={error.department}
                  >
                    <InputLabel id="demo-simple-select-outlined-label">Department</InputLabel>
                    <Select
                      id="department"
                      value={department}
                      onChange={this.change}
                      label="Department"
                    >
                      <MenuItem value="Fullstack">Full stack</MenuItem>
                      <MenuItem value="Frontend">Frontend</MenuItem>
                      <MenuItem value="Backend">Backend</MenuItem>
                    </Select>
                    {error.department && <FormHelperText>Please select department!</FormHelperText>}
                  </FormControl>
                </Grid>
              </Grid>
            </React.Fragment>
            <Button
              variant="contained"
              color="primary"
              onClick={this.handleSubmit}
              className={classes.button}
            >
              Submit
          </Button>
          </Paper>
        </main>
        <Container className={classes.cardGrid} maxWidth="md">
          <Grid container spacing={4}>
            {datas && this.getData('Fullstack')}
          </Grid>
        </Container>
        <Container className={classes.cardGrid} maxWidth="md">
          <Grid container spacing={4}>
            {datas && this.getData('Frontend')}
          </Grid>
        </Container>
        <Container className={classes.cardGrid} maxWidth="md">
          <Grid container spacing={4}>
            {datas && this.getData('Backend')}
          </Grid>
        </Container>
      </>
    );
  }
}

const styles = theme => ({
  layout: {
    width: 'auto',
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    [theme.breakpoints.up(600 + theme.spacing(2) * 2)]: {
      width: 600,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
  },
  paper: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(3),
    padding: theme.spacing(2),
    [theme.breakpoints.up(600 + theme.spacing(3) * 2)]: {
      marginTop: theme.spacing(6),
      marginBottom: theme.spacing(6),
      padding: theme.spacing(3),
    },
  },
  formControl: {
    minWidth: 200,
  },
  button: {
    marginTop: theme.spacing(3),
  },
  cardGrid: {
    paddingTop: theme.spacing(4),
  },
  card: {
    display: 'flex',
    flexDirection: 'column',
    margin: theme.spacing(1),
  },
  cardContent: {
    flexGrow: 1,
  },
  typography: {
    width: 'inherit',
  }
});

export default withStyles(styles)(App);
