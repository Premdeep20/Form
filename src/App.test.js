import React from 'react'
import { App } from './App';
import { shallow } from 'enzyme';
import TextField from '@material-ui/core/TextField';
import Select from '@material-ui/core/Select';
import Button from '@material-ui/core/Button';

const createTestProps = (prop) => ({
  ...prop,
});

let props;

describe('App', () => {

  beforeEach(() => {
    props = createTestProps({
      classes: {},
    });
  });

  it('should render App', () => {
    const wrapper = shallow(<App {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render App with employee data', () => {
    let res = JSON.stringify([
      {
        firstName: 'New',
        lastName: 'No',
        email: 'p@com',
        phoneNo: '1234567890',
        department: "Backend",
      }]);
    const spy = jest.spyOn(window.localStorage.__proto__, 'getItem');
    spy.mockImplementation(() => res)
    const wrapper = shallow(<App {...props} />);
    expect(wrapper).toMatchSnapshot();
    spy.mockRestore();
  });

  it('Check labels', () => {
    const wrapper = shallow(<App {...props} />);
    expect(wrapper.find(TextField).at(0).props().label).toEqual('First name');
    expect(wrapper.find(TextField).at(1).props().label).toEqual('Last name');
    expect(wrapper.find(TextField).at(2).props().label).toEqual('Email');
    expect(wrapper.find(TextField).at(3).props().label).toEqual('Phone no');
    expect(wrapper.find(Select).props().label).toEqual('Department');
  });

  it('should call change for first name', () => {
    const wrapper = shallow(<App {...props} />);
    wrapper.find(TextField).at(0).simulate('change', {
      target: {
        value: 'P',
        id: 'firstName',
      }
    });
    expect(wrapper.instance().state.data.firstName).toEqual('P');
  });

  it('should call change for last name', () => {
    const wrapper = shallow(<App {...props} />);
    wrapper.find(TextField).at(1).simulate('change', {
      target: {
        value: 'N',
        id: 'lastName',
      }
    });
    expect(wrapper.instance().state.data.lastName).toEqual('N');
  });

  it('should call change for email', () => {
    const wrapper = shallow(<App {...props} />);
    wrapper.find(TextField).at(2).simulate('change', {
      target: {
        value: 'p',
        id: 'email',
      }
    });
    expect(wrapper.instance().state.data.email).toEqual('p');
  });

  it('should call change for phone no', () => {
    const wrapper = shallow(<App {...props} />);
    wrapper.find(TextField).at(3).simulate('change', {
      target: {
        value: '9',
        id: 'phoneNo',
      }
    });
    expect(wrapper.instance().state.data.phoneNo).toEqual('9');
  });

  it('should call change for department', () => {
    const wrapper = shallow(<App {...props} />);
    wrapper.find(Select).simulate('change', {
      target: {
        value: 'Backend',
        id: 'department',
      }
    });
    expect(wrapper.instance().state.data.department).toEqual('Backend');
  });

  it('should call handleSubmit and do nothing as there is no data', () => {
    const wrapper = shallow(<App {...props} />);
    wrapper.find(Button).simulate('click', {});
    expect(wrapper.instance().state.data.firstName).toEqual('');
    expect(wrapper.instance().state.data.lastName).toEqual('');
    expect(wrapper.instance().state.data.email).toEqual('');
    expect(wrapper.instance().state.data.phoneNo).toEqual('');
    expect(wrapper.instance().state.data.department).toEqual('');
    expect(wrapper.instance().state.error.firstName).toBeTruthy();
    expect(wrapper.instance().state.error.lastName).toBeTruthy();
    expect(wrapper.instance().state.error.email).toBeTruthy();
    expect(wrapper.instance().state.error.phoneNo).toBeTruthy();
    expect(wrapper.instance().state.error.department).toBeTruthy();
  });

  it('should call handleSubmit store data in localstorage to existing data and clear form ', () => {
    let res = JSON.stringify([
      {
        firstName: 'New',
        lastName: 'No',
        email: 'p@com',
        phoneNo: '1234567890',
        department: "Backend",
      }]);
    const spy = jest.spyOn(window.localStorage.__proto__, 'getItem');
    spy.mockImplementation(() => res)
    const wrapper = shallow(<App {...props} />);
    wrapper.setState({
      data: {
        firstName: 'Premdeep',
        lastName: 'Singh',
        email: 'prem@yopmail.com',
        phoneNo: '1234567890',
        department: 'Frontend',
      }
    });
    wrapper.find(Button).simulate('click', {});
    expect(wrapper.instance().state.data.firstName).toEqual('');
    spy.mockRestore();
  });

  it('should call handleSubmit store data in localstorage to empty data and clear form ', () => {
    let res = JSON.stringify('');
    const spy = jest.spyOn(window.localStorage.__proto__, 'getItem');
    spy.mockImplementation(() => res)
    const wrapper = shallow(<App {...props} />);
    wrapper.setState({
      data: {
        firstName: 'Premdeep',
        lastName: 'Singh',
        email: 'prem@yopmail.com',
        phoneNo: '1234567890',
        department: 'Frontend',
      }
    });
    wrapper.find(Button).simulate('click', {});
    expect(wrapper.instance().state.data.firstName).toEqual('');
    spy.mockRestore();
  });

  it('should call handleSubmit and do nothing if error in first name', () => {
    let res = JSON.stringify('');
    const spy = jest.spyOn(window.localStorage.__proto__, 'getItem');
    spy.mockImplementation(() => res)
    const wrapper = shallow(<App {...props} />);
    wrapper.setState({
      data: {
        firstName: '',
        lastName: 'Singh',
        email: 'prem@yopmail.com',
        phoneNo: '1234567890',
        department: 'Frontend',
      }
    });
    wrapper.find(TextField).at(0).simulate('change', {
      target: {
        value: '9',
        id: 'firstName',
      }
    });
    wrapper.find(Button).simulate('click', {});
    expect(wrapper.instance().state.error.firstName).toBeTruthy();
    spy.mockRestore();
  });

  it('should call handleSubmit and do nothing if error in last name', () => {
    let res = JSON.stringify('');
    const spy = jest.spyOn(window.localStorage.__proto__, 'getItem');
    spy.mockImplementation(() => res)
    const wrapper = shallow(<App {...props} />);
    wrapper.setState({
      data: {
        firstName: 'Premdeep',
        lastName: '',
        email: 'prem@yopmail.com',
        phoneNo: '1234567890',
        department: 'Frontend',
      }
    });
    wrapper.find(TextField).at(1).simulate('change', {
      target: {
        value: '@',
        id: 'lastName',
      }
    });
    wrapper.find(Button).simulate('click', {});
    expect(wrapper.instance().state.error.lastName).toBeTruthy();
    spy.mockRestore();
  });

  it('should call handleSubmit and do nothing if error in email', () => {
    let res = JSON.stringify('');
    const spy = jest.spyOn(window.localStorage.__proto__, 'getItem');
    spy.mockImplementation(() => res)
    const wrapper = shallow(<App {...props} />);
    wrapper.setState({
      data: {
        firstName: 'Premdeep',
        lastName: 'Singh',
        email: 'prem@yopmail.',
        phoneNo: '1234567890',
        department: 'Frontend',
      }
    });
    wrapper.find(TextField).at(2).simulate('change', {
      target: {
        value: 'prem@yopmail.d',
        id: 'email',
      }
    });
    wrapper.find(Button).simulate('click', {});
    expect(wrapper.instance().state.error.email).toBeTruthy();
    spy.mockRestore();
  });

  it('should call handleSubmit and do nothing if error in phone n0', () => {
    let res = JSON.stringify('');
    const spy = jest.spyOn(window.localStorage.__proto__, 'getItem');
    spy.mockImplementation(() => res)
    const wrapper = shallow(<App {...props} />);
    wrapper.setState({
      data: {
        firstName: 'Premdeep',
        lastName: 'Singh',
        email: 'prem@yopmail.com',
        phoneNo: '955',
        department: 'Frontend',
      }
    });
    wrapper.find(TextField).at(3).simulate('change', {
      target: {
        value: '@',
        id: 'phoneNo',
      }
    });
    wrapper.find(Button).simulate('click', {});
    expect(wrapper.instance().state.error.phoneNo).toBeTruthy();
    spy.mockRestore();
  });
});