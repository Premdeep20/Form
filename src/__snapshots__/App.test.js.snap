// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App 1`] = `
<Fragment>
  <main>
    <WithStyles(ForwardRef(Paper))>
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
        variant="h6"
      >
        Employee details
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={3}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          sm={6}
          xs={12}
        >
          <WithStyles(ForwardRef(TextField))
            error={false}
            fullWidth={true}
            helperText={false}
            id="firstName"
            label="First name"
            name="firstName"
            onChange={[Function]}
            value=""
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          sm={6}
          xs={12}
        >
          <WithStyles(ForwardRef(TextField))
            error={false}
            fullWidth={true}
            helperText={false}
            id="lastName"
            label="Last name"
            name="lastName"
            onChange={[Function]}
            required={true}
            value=""
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(TextField))
            error={false}
            fullWidth={true}
            helperText={false}
            id="email"
            label="Email"
            name="email"
            onChange={[Function]}
            required={true}
            value=""
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          sm={6}
          xs={12}
        >
          <WithStyles(ForwardRef(TextField))
            error={false}
            fullWidth={true}
            helperText={false}
            id="phoneNo"
            label="Phone no"
            name="phoneNo"
            onChange={[Function]}
            required={true}
            value=""
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          sm={6}
          xs={12}
        >
          <WithStyles(ForwardRef(FormControl))
            error={false}
            fullWidth={true}
            required={true}
            variant="outlined"
          >
            <WithStyles(ForwardRef(InputLabel))
              id="demo-simple-select-outlined-label"
            >
              Department
            </WithStyles(ForwardRef(InputLabel))>
            <WithStyles(ForwardRef(Select))
              id="department"
              label="Department"
              onChange={[Function]}
              value=""
            >
              <WithStyles(ForwardRef(MenuItem))
                value="Fullstack"
              >
                Full stack
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                value="Frontend"
              >
                Frontend
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                value="Backend"
              >
                Backend
              </WithStyles(ForwardRef(MenuItem))>
            </WithStyles(ForwardRef(Select))>
          </WithStyles(ForwardRef(FormControl))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
        variant="contained"
      >
        Submit
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Paper))>
  </main>
  <WithStyles(ForwardRef(Container))
    maxWidth="md"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    />
  </WithStyles(ForwardRef(Container))>
  <WithStyles(ForwardRef(Container))
    maxWidth="md"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    />
  </WithStyles(ForwardRef(Container))>
  <WithStyles(ForwardRef(Container))
    maxWidth="md"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    />
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;

exports[`App should render App with employee data 1`] = `
<Fragment>
  <main>
    <WithStyles(ForwardRef(Paper))>
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
        variant="h6"
      >
        Employee details
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={3}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          sm={6}
          xs={12}
        >
          <WithStyles(ForwardRef(TextField))
            error={false}
            fullWidth={true}
            helperText={false}
            id="firstName"
            label="First name"
            name="firstName"
            onChange={[Function]}
            value=""
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          sm={6}
          xs={12}
        >
          <WithStyles(ForwardRef(TextField))
            error={false}
            fullWidth={true}
            helperText={false}
            id="lastName"
            label="Last name"
            name="lastName"
            onChange={[Function]}
            required={true}
            value=""
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(TextField))
            error={false}
            fullWidth={true}
            helperText={false}
            id="email"
            label="Email"
            name="email"
            onChange={[Function]}
            required={true}
            value=""
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          sm={6}
          xs={12}
        >
          <WithStyles(ForwardRef(TextField))
            error={false}
            fullWidth={true}
            helperText={false}
            id="phoneNo"
            label="Phone no"
            name="phoneNo"
            onChange={[Function]}
            required={true}
            value=""
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          sm={6}
          xs={12}
        >
          <WithStyles(ForwardRef(FormControl))
            error={false}
            fullWidth={true}
            required={true}
            variant="outlined"
          >
            <WithStyles(ForwardRef(InputLabel))
              id="demo-simple-select-outlined-label"
            >
              Department
            </WithStyles(ForwardRef(InputLabel))>
            <WithStyles(ForwardRef(Select))
              id="department"
              label="Department"
              onChange={[Function]}
              value=""
            >
              <WithStyles(ForwardRef(MenuItem))
                value="Fullstack"
              >
                Full stack
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                value="Frontend"
              >
                Frontend
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                value="Backend"
              >
                Backend
              </WithStyles(ForwardRef(MenuItem))>
            </WithStyles(ForwardRef(Select))>
          </WithStyles(ForwardRef(FormControl))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
        variant="contained"
      >
        Submit
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Paper))>
  </main>
  <WithStyles(ForwardRef(Container))
    maxWidth="md"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    />
  </WithStyles(ForwardRef(Container))>
  <WithStyles(ForwardRef(Container))
    maxWidth="md"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    />
  </WithStyles(ForwardRef(Container))>
  <WithStyles(ForwardRef(Container))
    maxWidth="md"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Typography))
        align="center"
        gutterBottom={true}
        variant="h5"
      >
        Backend
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={4}
        sm={6}
        xs={12}
      >
        <WithStyles(ForwardRef(Card))>
          <WithStyles(ForwardRef(CardContent))>
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h6"
            >
              New
               
              No
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              color="textSecondary"
            >
              p@com
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              color="textSecondary"
            >
              1234567890
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              color="textSecondary"
            >
              Backend
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(CardContent))>
        </WithStyles(ForwardRef(Card))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;
